Sources/OpenAPIRuntime/Interface/Body.swift

```diff
@@ -0,0 +1,535 @@
+    ///   than `maxBytes`.
+    public func collect(upTo maxBytes: Int) async throws -> DataType {
+
+        // As a courtesy, check if another iteration is allowed, and throw
+        // an error instead of fatalError here if the user is trying to
+        // iterate a sequence for the second time, if it's only safe to be
+        // iterated once.
+        if iterationBehavior == .single {
+            try {
+                lock.lock()
+                defer {
+                    lock.unlock()
+                }
+                guard !locked_iteratorCreated else {
+                    throw TooManyIterationsError()
+                }
+            }()
+        }
+
+        var buffer = DataType.init()
+        for try await chunk in self {
+            guard buffer.count + chunk.count <= maxBytes else {
+                throw TooManyBytesError(maxBytes: maxBytes)
+            }
+            buffer.append(contentsOf: chunk)
+        }
+        return buffer
+    }
+}
+
+// MARK: - String-based bodies
+
+extension StringProtocol {
+    fileprivate var asBodyChunk: Body.DataType {
+        Array(utf8)[...]
+    }
+}
+
+extension Body {
+
+    public convenience init(
+        data: some StringProtocol,
+        length: Length
+    ) {
+        self.init(
+            dataChunks: [data.asBodyChunk],
+            length: length
+        )
+    }
+
+    public convenience init(
+        data: some StringProtocol
+    ) {
+        self.init(
+            dataChunks: [data.asBodyChunk],
+            length: .known(data.count)
+        )
+    }
+
+    public convenience init<S: Sequence>(
+        dataChunks: S,
+        length: Length,
+        iterationBehavior: IterationBehavior
+    ) where S.Element: StringProtocol {
+        self.init(
+            dataChunks: dataChunks.map(\.asBodyChunk),
+            length: length,
+            iterationBehavior: iterationBehavior
+        )
+    }
+
+    public convenience init<C: Collection>(
+        dataChunks: C,
+        length: Length
+    ) where C.Element: StringProtocol {
+        self.init(
+            dataChunks: dataChunks.map(\.asBodyChunk),
+            length: length
+        )
+    }
+
+    public convenience init<C: Collection>(
+        dataChunks: C
+    ) where C.Element: StringProtocol {
+        self.init(
+            dataChunks: dataChunks.map(\.asBodyChunk)
+        )
+    }
+
+    public convenience init(
+        stream: AsyncThrowingStream<some StringProtocol, any Error>,
+        length: Body.Length
+    ) {
+        self.init(
+            sequence: .init(stream.map(\.asBodyChunk)),
+            length: length,
+            iterationBehavior: .single
+        )
+    }
+
+    public convenience init(
+        stream: AsyncStream<some StringProtocol>,
+        length: Body.Length
+    ) {
+        self.init(
+            sequence: .init(stream.map(\.asBodyChunk)),
+            length: length,
+            iterationBehavior: .single
+        )
+    }
```

Comment:
Same here as above.

Reply:
Same answer - `iterationBehavior: .single` is why this initializer is helpful.